azure_ai_search_rag.ipynb

```diff
@@ -0,0 +1,908 @@
+    "import os\n",
+    "\n",
+    "SEARCH_SERVICE_ENDPOINT = os.getenv(\"SEARCH_SERVICE_ENDPOINT\") or getpass(\"Enter your Azure AI Search Service Endpoint: \")\n",
+    "SEARCH_SERVICE_API_KEY = os.getenv(\"SEARCH_SERVICE_API_KEY\") or getpass(\"Enter your Azure AI Search Service API Key: \")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 72,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Using AAD for authentication.\n"
+     ]
+    }
+   ],
+   "source": [
+    "USE_AAD_FOR_SEARCH = True\n",
+    "\n",
+    "def authenticate_azure_search(api_key=None, use_aad_for_search=False):\n",
+    "    if use_aad_for_search:\n",
+    "        print(\"Using AAD for authentication.\")\n",
+    "        credential = DefaultAzureCredential()\n",
+    "    else:\n",
+    "        print(\"Using API keys for authentication.\")\n",
+    "        if api_key is None:\n",
+    "            raise ValueError(\"API key must be provided if not using AAD for authentication.\")\n",
+    "        credential = AzureKeyCredential(api_key)\n",
+    "    return credential\n",
+    "\n",
+    "azure_search_credential = authenticate_azure_search(\n",
+    "    api_key=SEARCH_SERVICE_API_KEY, use_aad_for_search=USE_AAD_FOR_SEARCH\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create a vector index"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 70,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      " ai-arxiv2-semantic-chunks created\n"
+     ]
+    }
+   ],
+   "source": [
+    "from azure.search.documents.indexes import SearchIndexClient\n",
+    "from azure.search.documents.indexes.models import (\n",
+    "    SimpleField,\n",
+    "    SearchFieldDataType,\n",
+    "    SearchableField,\n",
+    "    SearchField,\n",
+    "    VectorSearch,\n",
+    "    HnswAlgorithmConfiguration,\n",
+    "    VectorSearchProfile,\n",
+    "    SemanticConfiguration,\n",
+    "    SemanticPrioritizedFields,\n",
+    "    SemanticField,\n",
+    "    SemanticSearch,\n",
+    "    SearchIndex,\n",
+    ")\n",
+    "\n",
+    "DIMENSIONS = 1024\n",
+    "HNSW_PARAMETERS = {\"m\": 4, \"metric\": \"cosine\", \"ef_construction\": 400, \"ef_search\": 500}\n",
+    "INDEX_NAME = \"ai-arxiv2-semantic-chunks\"  # replace with your actual index name\n",
+    "\n",
+    "\n",
+    "def create_index(endpoint, credential):\n",
+    "    # Create a search index\n",
+    "    index_client = SearchIndexClient(endpoint=endpoint, credential=credential)\n",
+    "    fields = [\n",
+    "        SimpleField(\n",
+    "            name=\"id\",\n",
+    "            type=SearchFieldDataType.String,\n",
+    "            key=True,\n",
+    "            sortable=True,\n",
+    "            filterable=True,\n",
+    "            facetable=True,\n",
```

Comment:
Why mark this as facetable? Keys are not good candidates for this. Also, do you need it to be sortable?

Reply:
Updated